### PR TITLE
Switch container_pull & container_load to Go implementation

### PR DESCRIPTION
**Pull request by smukherj1**
Renamed python based rules to legacy_container_pull &
legacy_container_load

### CODE REVIEW

File: container/legacy_load.bzl [hunk 1]
```diff
@@ -11,57 +11,56 @@
...
 """
 
+load(
+    "//container:legacy_pull.bzl",
+    _python = "python",
+)
+
 def _impl(repository_ctx):
-    """Core implementation of new_container_load."""
+    """Core implementation of legacy_container_load."""
```
Comment (nlopezgi) [inline container/legacy_load.bzl]: add docs at top level of this file (and others for py container_pull) indicating this is deprecated and should not be modified unless there are breaking changes that need to be integrated

File: README.md [hunk 2]
```diff
@@ -1283,167 +1282,6 @@ Image references should not be updated individually because these images have
...
-           how to use <code>container_pull</code> after configuring the docker toolchain</p>
-        <p>When left unspecified (ie not set explicitly or set by the docker toolchain), docker will use
-        the directory specified via the DOCKER_CONFIG environment variable. If DOCKER_CONFIG isn't set,
-        docker falls back to $HOME/.docker.
-        </p>
-      </td>
-    </tr>
-  </tbody>
-</table>
-
```
Comment (nlopezgi) [inline README.md]: add puller_darwin and puller_linux docs somewhere down here

File: container/legacy_load.bzl (hunk 1 above)
Comment (smukherj1) [reply container/legacy_load.bzl]: done

File: README.md (hunk 2 above)
Comment (smukherj1) [reply README.md]: done

File: README.md [hunk 3]
```diff
@@ -1519,6 +1527,22 @@ bazel startup flag `--loading_phase_threads=1` in your bazel invocation.
            include CPU features such as <code>["sse4", "aes"]</code>.</p>
       </td>
     </tr>
+    <tr>
+      <td><code>puller_darwin</code></td>
+      <td>
+        <p><code>label; optional</code></p>
+        <p>A Mac 64-bit binary that implements the functionality provided by
```
Comment (nlopezgi) [inline README.md]: nit - add: "Visible for testing purposes" here and below

Comment (smukherj1) [reply README.md]: Done

**Pull request merged**